esinstall/src/index.ts

```diff
@@ -333,6 +335,13 @@ export async function install(
             importMap.imports[key] = `./${targetName}.js`;
           });
       } else if (resolvedResult.type === 'ASSET') {
+        // add extension if missing
+        const isMissingExt = path.extname(resolvedResult.loc) && !path.extname(proxiedName);
+        if (isMissingExt) {
+          const ext = path.basename(resolvedResult.loc).replace(/[^.]+/, '');
+          targetName += ext;
+          proxiedName += ext;
+        }
```

Comment:
This fixes another bug where for non-JS assets, the extension was often stripped (e.g. when installing `bulma`, esinstall would create an extensionless `web_modules/bulma` file that was supposed to be `web_modules/bulma.sass`.

While this probably isn’t a perfect solution, I think it’s better, as it more closely matches the JS behavior. Also of note: we don’t add extensions if there aren’t any.

Reply:
the thinking here makes sense, but `resolvedResult.loc` should already be a fully resolved file location. If we want to fix/handle this, the fix should live inside of `resolveWebDependency()`

Reply:
So `resolvedResult.loc` is correct, and there‘s nothing to change there. That‘s why we correctly get the extension from there.

I was just making a change here where `getWebDependencyName()` and `sanitizePackageName()` were producing unexpected behavior for non-JS assets. Because we were adding asset handling here, I felt it safer to make the change here than in those 2 functions. But happy for you to make the call.

Reply:
got it, makes sense!